service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java

```diff
@@ -488,7 +488,7 @@ public HeartbeatResponse heartbeat(String microserviceId, String microserviceIns
     RestUtils.put(ipPort,
         String.format(Const.REGISTRY_API.MICROSERVICE_HEARTBEAT, microserviceId, microserviceInstanceId),
         new RequestParam().setTimeout(ServiceRegistryConfig.INSTANCE.getHeartBeatRequestTimeout()),
-        syncHandler(countDownLatch, HttpClientResponse.class, holder));
+        syncHandlerForHeartbeat(countDownLatch, HttpClientResponse.class, holder));
```

Comment:
Do not duplicate this code as mentioned above.

Reply:
ok. i fix it